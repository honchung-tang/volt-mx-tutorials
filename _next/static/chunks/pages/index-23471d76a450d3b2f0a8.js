_N_E=(window.webpackJsonp_N_E=window.webpackJsonp_N_E||[]).push([[45],{"8SHQ":function(n,t,c){"use strict";c.d(t,"f",(function(){return o})),c.d(t,"c",(function(){return u})),c.d(t,"a",(function(){return r})),c.d(t,"b",(function(){return e})),c.d(t,"d",(function(){return i})),c.d(t,"e",(function(){return s}));var o=!1,u=3200,r="master",e="volt-mx-tutorials",i="HCL-TECH-SOFTWARE",s=o?"http://localhost:".concat(u):"".concat("https://opensource.hcltechsw.com","/").concat(e)},RNiq:function(n,t,c){"use strict";c.r(t);var o=c("q1tI"),u=c("20a2"),r=c("8SHQ");t.default=function(){var n=Object(u.useRouter)();return Object(o.useEffect)((function(){var t=r.f?"/hikes":"/".concat(r.b,"/hikes");n.push(t,void 0,{shallow:!0})})),null}},vlRD:function(n,t,c){(window.__NEXT_P=window.__NEXT_P||[]).push(["/",function(){return c("RNiq")}])}},[["vlRD",1,2,0]]]);