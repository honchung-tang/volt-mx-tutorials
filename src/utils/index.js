import _startCase from 'lodash/startCase';
import _toLower from 'lodash/toLower';
import moment from 'moment';

/**
 *  Truncates the string.
 * @private
 * @param {string} string - The string to truncate.
 * @param {number} length - The length of truncated string.
 * @returns {string} Returns truncated `string`.
 */

export const truncate = (string, length) => {
  if (typeof length !== 'number' || typeof string !== 'string') return string;
  if (string.length <= length) return string;
  if (length <= 3) return `${string.slice(0, length)}...`;
  return `${string.slice(0, length - 3)}...`;
};

/**
 *  Truncates the string without cutting words.
 * @private
 * @param {string} string - The string to truncate.
 * @param {number} length - The length of truncated string.
 * @returns {string} Returns truncated `string`.
 */

export const truncateAdvance = (string, length) => {
  if (typeof length !== 'number' || typeof string !== 'string') return string;
  if (string.length <= length) return string;
  return `${string.replace(new RegExp(`^(.{${length}}[^\\s]*).*`), '$1')}`;
};

/**
 *  Capitalize First Letter of the string.
 * @private
 * @param {string} string - The string whose first letter to be capitalized.
 * @returns {string} Returns string whose first letter is capital.
 */

export const capitalizeFirstLetter = (string) => {
  if (!string) {
    return null;
  }
  return _startCase(_toLower(string));
};

/**
 *  Get month name from the timestamp.
 * @private
 * @param {number} timestamp - The timestamp which is going to converted.
 * @returns {string} Returns month name from timestamp passed.
 */

export const getMonthNameFromTimestamp = (timestamp) => {
  const date = new Date(timestamp);
  const monthName = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];
  return monthName[date.getMonth()];
};

/**
 *  Get device os platform.
 * @private
 * @returns {string} Returns name of platform.
 */

let os = null;
if (process.browser) {
  const { userAgent, platform } = window.navigator;
  const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
  const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'];
  const iosPlatforms = ['iPhone', 'iPad', 'iPod', 'iPhone Simulator', 'iPod Simulator', 'iPad Simulator'];
  if (macosPlatforms.indexOf(platform) !== -1) {
    os = 'Mac OS';
  } else if (iosPlatforms.indexOf(platform) !== -1) {
    os = 'iOS';
  } else if (windowsPlatforms.indexOf(platform) !== -1) {
    os = 'Windows';
  } else if (/Android/.test(userAgent)) {
    os = 'Android';
  } else if (!os && /Linux/.test(platform)) {
    os = 'Linux';
  }
}

export const getOS = os;

/**
 * Load a script.
 * @private
 * @param src of the file
 */

export const loadScript = (src) => {
  const tag = document.createElement('script');
  tag.src = src;
  tag.async = true;
  document.body.appendChild(tag);
};

export const validateData = (d) => {
  if (d) {
    if (typeof (d) === 'string' && d !== '') {
      return true;
    }
    if (typeof (d) === 'object' && Object.keys(d).length > 0) {
      return true;
    }
  }
  return false;
};

const escapeRegExp = string => string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');

/**
 * Replace all occurences of string with other.
 *
 * @private
 *
 * @param str, String to be replaced on.
 * @param term, term to be replaced.
 * @param replacement, String to be replaced with.
 */
export const replaceAll = (str, term, replacement) => str.replace(new RegExp(escapeRegExp(term), 'g'), replacement);

/**
 * Convert number into two digits.
 *
 * @private
 *
 * @param number, Number to be converted.
 *
 * @return String, Number in two digits.
 */
function numberToTwoDigits(number) {
  return number > 9 ? `${number}` : `0${number}`;
}

/**
 * Convert seconds into HH:mm:ss time format.
 *
 * @private
 *
 * @param second, Number of seconds.
 *
 * @return String, time in the HH:mm:ss format.
 */
export const secondsToTime = (second) => {
  const seconds = parseInt(second, 10);
  const momentDuration = moment.duration(seconds, 'seconds');
  const hours = numberToTwoDigits(Math.floor(momentDuration.asHours()));
  const mins = numberToTwoDigits(momentDuration.minutes());
  const secs = numberToTwoDigits(momentDuration.seconds());
  return `${hours}:${mins}:${secs}`;
};
