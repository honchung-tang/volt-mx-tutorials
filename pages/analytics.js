import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Row from 'antd/lib/row';
import Col from 'antd/lib/col';
import moment from 'moment';
import AnalyticsHeader from '../src/components/AnalyticsHeader';
import { marketplaceActions } from '../src/actions';
import initialize, { instance as axios } from '../src/utils/initialize';
import KonyLayout from '../src/components/KonyLayout';
import DateRangePicker from '../src/components/DateRangePicker';
import KonyLoader from '../src/components/KonyLoader';
import style from './style.scss';
import PageNotFound from '../src/components/PageNotFound';
import KonyButton from '../src/components/KonyButton';
import AnalyticsMiniPanel from '../src/components/AnalyticsMiniPanel';
import AnalyticsPanel from '../src/components/AnalyticsPanel';

class AnalyticsPage extends Component {
  constructor(props) {
    super(props);
    const today = AnalyticsPage.getFormattedDate(new Date());
    const lastWeek = AnalyticsPage
      .getFormattedDate(
        new Date(new Date().getTime() - 7 * 24 * 60 * 60 * 1000),
      );

    this.state = {
      data: [],
      loading: false,
      dates: [lastWeek, today],
      momentDates: [moment(lastWeek), moment(today)],
    };
    this.setDate = this.setDate.bind(this);
    this.getData = this.getData.bind(this);
  }

  static getFormattedDate(date) {
    const today = date;
    const dd = String(today.getDate()).padStart(2, '0');
    const mm = String(today.getMonth() + 1).padStart(2, '0'); // January is 0!
    const yyyy = today.getFullYear();
    return `${yyyy}-${mm}-${dd}`;
  }

  static async getInitialProps(ctx) {
    await initialize(ctx);
    await ctx.store.dispatch(marketplaceActions.getConfig(true));
  }

  componentDidMount() {
    const { dates } = this.state;
    this.getData(dates);
  }

  getData = async (dates) => {
    try {
      this.setState({ loading: true });
      const res = await axios.get('api/v1_0/marketplace/get_analytics_data_dashboard', {
        params: {
          from_date: dates[0],
          to_date: dates[1],
        },
      });
      this.setState({ loading: false, data: res.data });
    } catch (error) {
      this.setState({ loading: false, data: [] });
    }
  };

  setDate(dates, momentDates) {
    this.setState({ dates, momentDates });
    this.getData(dates);
  }

  renderContent = (isModerator) => {
    const {
      loading, dates, momentDates, data,
    } = this.state;
    if (isModerator) {
      if (loading) {
        return <KonyLoader />;
      }
      return (
        <Fragment>
          <Row className={style.panelContainer}>
            <Col span={12} className={style.datePanel}>
              <KonyButton
                title="Component Performance"
                type="action"
                color="blue"
                href="/analytics/component"
                target="_blank"
                className={style.componentPerformanceButton}
              />
              <KonyButton
                title="Search Analytics"
                type="action"
                color="blue"
                href="/analytics/search"
                target="_blank"
              />
            </Col>
            <Col span={12} className={style.datePanel}>
              <DateRangePicker
                setDate={this.setDate}
                defaultDateValuesInMoment={momentDates}
                defaultDateValues={dates}
              />
            </Col>
          </Row>
          <Row>
            <Row className={style.miniPanelContainer}>
              <Col span={8}>
                <AnalyticsMiniPanel title="Total Number Of Components" count={data.totalAssets} icon="component" className="totalComponentCount"/>
              </Col>
              <Col span={8}>
                <AnalyticsMiniPanel title="Total Number Of Downloads" count={data.totalDownloads} icon="download" className="totalDownloadsCount"/>
              </Col>
              <Col span={8}>
                <AnalyticsMiniPanel title="Total Number Of Imports to Viz Forms" count={data.totalVizImports} icon="import" className="totalVizFormsImportsCount"/>
              </Col>
            </Row>
            <Row>
              <Col span={12}>
                <AnalyticsPanel title="Most Downloaded Components" data={data.topFiveDownloads} className="topFiveDownloadsCount"/>
              </Col>
              <Col span={12}>
                <AnalyticsPanel title="Most Deployed Components" data={data.topFiveDeployments} className="topFiveDeployedCount"/>
              </Col>
            </Row>
          </Row>
        </Fragment>
      );
    }

    return <PageNotFound />;
  };

  render() {
    const { marketplace, url } = this.props;
    const { config, moderatorMenu } = marketplace;
    return (
      <KonyLayout config={config} url={url}>
        <AnalyticsHeader placeholder="Search for Marketplace Assets" />
        <div>
          {
            this.renderContent(moderatorMenu
              && moderatorMenu.subMenu
              && moderatorMenu.subMenu.length > 0)
          }
        </div>
      </KonyLayout>
    );
  }
}

const mapStateToProps = ({ marketplace }) => ({ marketplace });

const mapDispatchToProps = dispatch => bindActionCreators({ ...marketplaceActions }, dispatch);
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AnalyticsPage);
