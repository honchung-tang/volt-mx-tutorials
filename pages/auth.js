import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { oauthActions } from '../src/actions';
import KonyLoader from '../src/components/KonyLoader';

class Auth extends Component {
  componentDidMount() {
    const query = this.props.url.query;
    if (process.browser) {
      this.props.login(query);
    }
  }

  render() {
    return <KonyLoader />;
  }
}

function mapStateToProps({ marketplace, assets }) {
  return {
    mp: marketplace,
    assets
  };
}
const mapDispatchToProps = dispatch => bindActionCreators({ ...oauthActions }, dispatch);
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Auth);
