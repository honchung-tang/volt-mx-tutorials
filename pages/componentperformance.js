import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Row from 'antd/lib/row';
import Col from 'antd/lib/col';
import _filter from 'lodash/filter';
import _forEach from 'lodash/forEach';
import Router from 'next/router';
import moment from 'moment';
import AnalyticsHeader from '../src/components/AnalyticsHeader';
import { marketplaceActions } from '../src/actions';
import initialize, { instance as axios } from '../src/utils/initialize';
import KonyLayout from '../src/components/KonyLayout';
import KonyBreadcrumb from '../src/components/KonyBreadcrumb';
import DateRangePicker from '../src/components/DateRangePicker';
import style from './style.scss';
import PageNotFound from '../src/components/PageNotFound';
import AnalyticsFilterTable from '../src/components/AnalyticsFilterTable';
import ExportTableToCSV from '../src/components/ExportTableToCSV';
import { baseURL } from '../src/config/settings';

export const columns = [
  {
    title: 'Comp. Name',
    dataIndex: 'component',
    sorter: (a, b) => a.component.localeCompare(b.component),
    sortDirections: ['descend', 'ascend'],
    width: 110,
  },
  {
    title: 'Mp Downloads',
    dataIndex: 'Downloads',
    sorter: (a, b) => a.Downloads - b.Downloads,
    width: 100,
  },
  {
    title: 'Imports into Forms',
    dataIndex: 'overallUsage',
    sorter: (a, b) => a.overallUsage - b.overallUsage,
    sortDirections: ['descend', 'ascend'],
    width: 100,
  },
  {
    title: 'Went Live',
    dataIndex: 'liveUsage',
    sorter: (a, b) => a.liveUsage - b.liveUsage,
    sortDirections: ['descend', 'ascend'],
    width: 100,
  },
  {
    title: 'No. of Debug',
    dataIndex: 'debug',
    sorter: (a, b) => a.debug - b.debug,
    sortDirections: ['descend', 'ascend'],
    width: 100,
  },
  {
    title: 'No. of releases',
    dataIndex: 'release',
    sorter: (a, b) => a.release - b.release,
    sortDirections: ['descend', 'ascend'],
    width: 100,
  },
  {
    title: 'Viz / Downloads',
    dataIndex: 'vizDownloads',
    sorter: (a, b) => a.vizDownloads - b.vizDownloads,
    sortDirections: ['descend', 'ascend'],
    width: 100,
  },
  {
    title: 'Live / Viz',
    dataIndex: 'liveViz',
    sorter: (a, b) => a.liveViz - b.liveViz,
    sortDirections: ['descend', 'ascend'],
    width: 80,
  },
  {
    title: 'Project vs Hits',
    dataIndex: 'output',
    width: 200,
  },
];

export class ComponentPerformancePage extends Component {
  constructor(props) {
    super(props);
    const today = ComponentPerformancePage.getFormattedDate(new Date());
    const lastWeek = ComponentPerformancePage
      .getFormattedDate(
        new Date(new Date().getTime() - 7 * 24 * 60 * 60 * 1000),
      );

    this.state = {
      data: [],
      originalData: [],
      loading: false,
      dates: [lastWeek, today],
      momentDates: [moment(lastWeek), moment(today)],
    };
  }

  static getFormattedDate(date) {
    const today = date;
    const dd = String(today.getDate()).padStart(2, '0');
    const mm = String(today.getMonth() + 1).padStart(2, '0'); // January is 0!
    const yyyy = today.getFullYear();
    return `${yyyy}-${mm}-${dd}`;
  }

  static async getInitialProps(ctx) {
    await initialize(ctx);
    await ctx.store.dispatch(marketplaceActions.getConfig(true));
  }

  componentDidMount() {
    const { dates } = this.state;
    this.getData(dates);
  }

  getTokenizedString(item) {
    const { output } = item;
    const splitOutput = output.split(',');
    splitOutput.pop();
    let results = '';
    _forEach(splitOutput, (splitItem) => {
      const splits = splitItem.split('/');
      if (splits.length > 1) {
        const result = `${splits[0]} ${splits[1].split(':')[1]}`;
        results += `${result} , `;
      } else if (splits.length === 1) {
        const result = splits[0].replace(/:/g, ' ');
        results += `${result} , `;
      }
    });
    results = results.substring(0, results.length - 2);
    return results;
  }

  getData = async (dates) => {
    try {
      this.setState({ loading: true });
      const res = await axios.get('api/v1_1/marketplace/analytics', {
        params: {
          from_date: dates[0],
          to_date: dates[1],
        },
      });
      const { data } = res;
      const originalData = data.map(item => ({ ...item, output: this.getTokenizedString(item) }));
      this.setState({ loading: false, data: originalData, originalData });
    } catch (error) {
      this.setState({ loading: false, data: [], originalData: [] });
    }
  };

  setDate = (dates, momentDates) => {
    this.setState({ dates, momentDates });
    this.getData(dates);
  }

  filterData = (keyword) => {
    const { originalData } = this.state;
    this.setState(
      {
        data: _filter(originalData,
          obj => obj
            .component
            .toString()
            .toLowerCase()
            .includes(keyword
              .toString()
              .toLowerCase())),
      },
    );
  }

  onRowClickParent = (data) => {
    const { dates } = this.state;
    const query = {
      data: encodeURIComponent(JSON.stringify(data)),
      from_date: dates[0],
      to_date: dates[1],
    };
    Router.replace({
      pathname: '/analyticsdetails',
      query,
    });
  };

  onChange = (pagination, filters, sorter, extra) => {
    const data = extra.currentDataSource;
    this.setState({ loading: false, data, originalData: data });
  };

  renderContent = (isModerator) => {
    const {
      loading, dates, momentDates, data,
    } = this.state;

    if (isModerator) {
      return (
        <Fragment>
          <Row>
            <Col span={14}>
              <KonyBreadcrumb
                title="Component Performance"
                subRoutes={
                  [
                    {
                      name: 'Dashboard',
                      path: `${baseURL}marketplace/dashboard`,
                    },
                    {
                      name: 'Analytics',
                      path: '/analytics',
                    },

                  ]
                }
              />
            </Col>
            <Col span={10} className={style.datePanel}>
              <DateRangePicker
                setDate={this.setDate}
                defaultDateValuesInMoment={momentDates}
                defaultDateValues={dates}
              />
            </Col>
          </Row>
          <Row>
            <Col>
              <ExportTableToCSV
                data={data}
                columns={columns}
                fileNameSuffix="components_performance_analytics"
              />
            </Col>
            <Col className={style.filterTablePanel}>
              <AnalyticsFilterTable
                onRowClick={this.onRowClickParent}
                showPagination
                data={data}
                onChange={this.onChange}
                loading={loading}
                columns={columns}
              />
            </Col>
          </Row>
        </Fragment>
      );
    }

    return <PageNotFound />;
  };

  render() {
    const { marketplace, url } = this.props;
    const { config, moderatorMenu } = marketplace;
    return (
      <KonyLayout config={config} url={url}>
        <AnalyticsHeader
          filterData={this.filterData}
          placeholder="Search for Marketplace Assets"
          search
        />
        <div>
          {
            this.renderContent(moderatorMenu
              && moderatorMenu.subMenu
              && moderatorMenu.subMenu.length > 0)
          }
        </div>
      </KonyLayout>
    );
  }
}

ComponentPerformancePage.propTypes = {
  marketplace: PropTypes.shape({}).isRequired,
  url: PropTypes.string.isRequired,
};

const mapStateToProps = ({ marketplace }) => ({ marketplace });

const mapDispatchToProps = dispatch => bindActionCreators({ ...marketplaceActions }, dispatch);
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ComponentPerformancePage);
