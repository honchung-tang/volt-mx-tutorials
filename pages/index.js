import React,{ useEffect } from 'react';
import { useRouter } from 'next/router';
import { BASE_PATH_URL, isDev } from '../src/config';

const MpLanding = () => {
  const router = useRouter()

  useEffect(() => {
    // redirect to /hikes route
    const routePath = isDev ? '/hikes' : `/${BASE_PATH_URL}/hikes`
    router.push(routePath, undefined, { shallow: true })
  })
  
  return null;
}

export default MpLanding
