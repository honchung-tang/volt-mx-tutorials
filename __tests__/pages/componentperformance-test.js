import React from 'react';
import { ComponentPerformancePage, columns } from '../../pages/componentperformance';
import { shallow } from '../../src/utils/enzyme';

describe('ComponentPerformancePage page test suite', () => {
  it('static methods', () => {
    const testDate = new Date('Fri Oct 03 2019');
    const expectedDate = '2019-10-03';
    expect(ComponentPerformancePage.getFormattedDate(testDate)).toBe(expectedDate);
    expect(ComponentPerformancePage.getTokenizedString({ output: 'keyword,bread,crumb' })).toBe('keyword , bread ');
    expect(ComponentPerformancePage.getTokenizedString({ output: 'bread/crumb:keyword, word' })).toBe('bread keyword ');
    expect(ComponentPerformancePage.getTokenizedString({ output: 'keyword,bread/crumb' })).toBe('keyword ');

    expect(columns[0].sorter({ component: 'ab' }, { component: 'bc' })).toBe(-1);
    expect(columns[0].sorter({ component: 'ab' }, { component: 'ab' })).toBe(0);
    columns.forEach((column) => {
      const { dataIndex } = column;
      if (dataIndex !== 'component' && dataIndex !== 'output') {
        const a = {};
        const b = {};
        a[dataIndex] = 40;
        b[dataIndex] = 10;
        expect(column.sorter(a, b)).toBe(30);
        a[dataIndex] = 10;
        b[dataIndex] = 40;
        expect(column.sorter(a, b)).toBe(-30);
      }
    });
  });
  it('render JSX', () => {
    const wrappedComponent = shallow(<ComponentPerformancePage marketplace={{}} url="/" />);

    expect(wrappedComponent).toMatchSnapshot();
  });
  it('render JSX with moderator menu', () => {
    const wrappedComponent = shallow(<ComponentPerformancePage
      marketplace={{
        moderatorMenu: {},
      }}
      url="/"
    />);

    expect(wrappedComponent).toMatchSnapshot();
  });
  it('render JSX with moderator sub menu', () => {
    const wrappedComponent = shallow(<ComponentPerformancePage
      marketplace={{
        config: {},
        moderatorMenu: {
          subMenu: [{}],
        },
      }}
      url="/"
    />);

    expect(wrappedComponent).toMatchSnapshot();
  });
  it('render JSX with moderator Sub Menu and call setDate and filterData', () => {
    const wrappedComponent = shallow(<ComponentPerformancePage
      marketplace={{
        config: {},
        moderatorMenu: {
          subMenu: [{}],
        },
      }}
      url="/"
    />);
    wrappedComponent.instance().setDate();
    wrappedComponent.instance().filterData('keyword');
    expect(wrappedComponent).toMatchSnapshot();
  });
});
