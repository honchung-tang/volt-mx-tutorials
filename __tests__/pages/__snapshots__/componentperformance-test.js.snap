// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentPerformancePage page test suite render JSX 1`] = `
<Connect(KonyLayout)
  url="/"
>
  <Connect(AnalyticsHeader)
    filterData={[Function]}
    placeholder="Search for Marketplace Assets"
    search={true}
  />
  <div>
    <pageNotFound />
  </div>
</Connect(KonyLayout)>
`;

exports[`ComponentPerformancePage page test suite render JSX with moderator Sub Menu and call setDate and filterData 1`] = `
<Connect(KonyLayout)
  config={Object {}}
  url="/"
>
  <Connect(AnalyticsHeader)
    filterData={[Function]}
    placeholder="Search for Marketplace Assets"
    search={true}
  />
  <div>
    <Row
      gutter={0}
    >
      <Col
        span={14}
      >
        <KonyBreadcrumb
          rootRedirect={null}
          subRoutes={
            Array [
              Object {
                "name": "Dashboard",
                "path": "http://community.qa-kony.com/marketplace/dashboard",
              },
              Object {
                "name": "Analytics",
                "path": "/analytics",
              },
            ]
          }
          title="Component Performance"
        />
      </Col>
      <Col
        className="datePanel"
        span={10}
      >
        <DateRangePicker
          setDate={[Function]}
        />
      </Col>
    </Row>
    <Row
      gutter={0}
    >
      <Col>
        <ExportTableToCSV
          columns={
            Array [
              Object {
                "dataIndex": "component",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Comp. Name",
                "width": 110,
              },
              Object {
                "dataIndex": "Downloads",
                "defaultSortOrder": "descend",
                "sorter": [Function],
                "title": "Mp Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "overallUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Imports into Forms",
                "width": 100,
              },
              Object {
                "dataIndex": "liveUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Went Live",
                "width": 100,
              },
              Object {
                "dataIndex": "debug",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of Debug",
                "width": 100,
              },
              Object {
                "dataIndex": "release",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of releases",
                "width": 100,
              },
              Object {
                "dataIndex": "vizDownloads",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Viz / Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "liveViz",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Live / Viz",
                "width": 80,
              },
              Object {
                "dataIndex": "output",
                "title": "Project vs Hits",
                "width": 200,
              },
            ]
          }
          data={Array []}
          fileNameSuffix="components_performance_analytics"
          title="Export as Excel"
        />
      </Col>
      <Col
        className="filterTablePanel"
      >
        <AnalyticsFilterTable
          columns={
            Array [
              Object {
                "dataIndex": "component",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Comp. Name",
                "width": 110,
              },
              Object {
                "dataIndex": "Downloads",
                "defaultSortOrder": "descend",
                "sorter": [Function],
                "title": "Mp Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "overallUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Imports into Forms",
                "width": 100,
              },
              Object {
                "dataIndex": "liveUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Went Live",
                "width": 100,
              },
              Object {
                "dataIndex": "debug",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of Debug",
                "width": 100,
              },
              Object {
                "dataIndex": "release",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of releases",
                "width": 100,
              },
              Object {
                "dataIndex": "vizDownloads",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Viz / Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "liveViz",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Live / Viz",
                "width": 80,
              },
              Object {
                "dataIndex": "output",
                "title": "Project vs Hits",
                "width": 200,
              },
            ]
          }
          data={Array []}
          onRowClick={[Function]}
          showPagination={true}
        />
      </Col>
    </Row>
  </div>
</Connect(KonyLayout)>
`;

exports[`ComponentPerformancePage page test suite render JSX with moderator Sub Menu and call setDate and filterData 2`] = `
<Connect(KonyLayout)
  config={Object {}}
  url="/"
>
  <Connect(AnalyticsHeader)
    filterData={[Function]}
    placeholder="Search for Marketplace Assets"
    search={true}
  />
  <div>
    <Row
      gutter={0}
    >
      <Col
        span={14}
      >
        <KonyBreadcrumb
          rootRedirect={null}
          subRoutes={
            Array [
              Object {
                "name": "Dashboard",
                "path": "http://community.qa-kony.com/marketplace/dashboard",
              },
              Object {
                "name": "Analytics",
                "path": "/analytics",
              },
            ]
          }
          title="Component Performance"
        />
      </Col>
      <Col
        className="datePanel"
        span={10}
      >
        <DateRangePicker
          setDate={[Function]}
        />
      </Col>
    </Row>
    <Row
      gutter={0}
    >
      <Col>
        <ExportTableToCSV
          columns={
            Array [
              Object {
                "dataIndex": "component",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Comp. Name",
                "width": 110,
              },
              Object {
                "dataIndex": "Downloads",
                "defaultSortOrder": "descend",
                "sorter": [Function],
                "title": "Mp Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "overallUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Imports into Forms",
                "width": 100,
              },
              Object {
                "dataIndex": "liveUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Went Live",
                "width": 100,
              },
              Object {
                "dataIndex": "debug",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of Debug",
                "width": 100,
              },
              Object {
                "dataIndex": "release",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of releases",
                "width": 100,
              },
              Object {
                "dataIndex": "vizDownloads",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Viz / Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "liveViz",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Live / Viz",
                "width": 80,
              },
              Object {
                "dataIndex": "output",
                "title": "Project vs Hits",
                "width": 200,
              },
            ]
          }
          data={Array []}
          fileNameSuffix="components_performance_analytics"
          title="Export as Excel"
        />
      </Col>
      <Col
        className="filterTablePanel"
      >
        <AnalyticsFilterTable
          columns={
            Array [
              Object {
                "dataIndex": "component",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Comp. Name",
                "width": 110,
              },
              Object {
                "dataIndex": "Downloads",
                "defaultSortOrder": "descend",
                "sorter": [Function],
                "title": "Mp Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "overallUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Imports into Forms",
                "width": 100,
              },
              Object {
                "dataIndex": "liveUsage",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Went Live",
                "width": 100,
              },
              Object {
                "dataIndex": "debug",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of Debug",
                "width": 100,
              },
              Object {
                "dataIndex": "release",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "No. of releases",
                "width": 100,
              },
              Object {
                "dataIndex": "vizDownloads",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Viz / Downloads",
                "width": 100,
              },
              Object {
                "dataIndex": "liveViz",
                "sortDirections": Array [
                  "descend",
                  "ascend",
                ],
                "sorter": [Function],
                "title": "Live / Viz",
                "width": 80,
              },
              Object {
                "dataIndex": "output",
                "title": "Project vs Hits",
                "width": 200,
              },
            ]
          }
          data={Array []}
          onRowClick={[Function]}
          showPagination={true}
        />
      </Col>
    </Row>
  </div>
</Connect(KonyLayout)>
`;

exports[`ComponentPerformancePage page test suite render JSX with moderator menu 1`] = `
<Connect(KonyLayout)
  url="/"
>
  <Connect(AnalyticsHeader)
    filterData={[Function]}
    placeholder="Search for Marketplace Assets"
    search={true}
  />
  <div>
    <pageNotFound />
  </div>
</Connect(KonyLayout)>
`;

exports[`ComponentPerformancePage page test suite render JSX with moderator sub menu 1`] = `
<Connect(KonyLayout)
  config={Object {}}
  url="/"
>
  <Connect(AnalyticsHeader)
    filterData={[Function]}
    placeholder="Search for Marketplace Assets"
    search={true}
  />
  <div>
    <KonyLoader
      inline={false}
    />
  </div>
</Connect(KonyLayout)>
`;
